src/internal/HttpRequest.ts

```diff
@@ -18,12 +18,6 @@ function getURL(options: HttpRequestOptions) {
     url.port = options.port.toString();
     url.pathname = options.path;
 
-    if (options.auth !== undefined) {
-        const [user, pass] = options.auth.split(':');
-        url.username = user;
-        url.password = pass;
-    }
-
```

Comment:
co se stane kdyz bude jmeno nebo heslo obsahovat nepovolene znaky (# nebo : v hesle) je potreba to mit spravne osetrene